File: src/main/java/com/roamy/web/CityResource.java
package com.roamy.web;

import com.roamy.domain.City;
import com.roamy.service.api.CityService;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Abhijit on 10/14/2015.
 */
@RestController
@RequestMapping(value = "/cities")
public class CityResource {

    private static final Logger logger = LoggerFactory.getLogger(CityResource.class);

    @Autowired
    private CityService cityService;

    @RequestMapping(value = "/{cityId}", method = RequestMethod.GET)
    public City getCity(@PathVariable Long cityId) {

        City city = cityService.getCityById(cityId);
        logger.info("Found {}", city);

        return city;
    }

    @RequestMapping(method = RequestMethod.GET)
    public List<City> getAllCities() {

        List<City> cities = cityService.getAllActiveCities();
        logger.info("Found {} cities", cities == null ? 0 : cities.size());

        return cities;
    }
}

File: src/main/scala/com/roamy/web/CityResource.scala
package com.roamy.web

import com.google.gson.{Gson, JsonObject}
import com.roamy.AllPassRestApiEntitlementsFilter
import com.roamy.dao.api.CityRepository
import com.roamy.domain.City
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.web.bind.annotation.{RequestMapping, RestController}

/**
 * Created by amit on 10/17/15.
 */
@RestController
@RequestMapping(Array("/cities"))
class CityResource extends RestApiEndPoint[City] {

  val entitlementsFilter = new AllPassRestApiEntitlementsFilter[City] {}


  @Autowired
  var cityRepository: CityRepository = _


  def repository = {
    this.cityRepository.asInstanceOf[JpaRepository[City, java.lang.Long]]
  }

}
